dm/syncer/syncer.go

```diff
@@ -2795,6 +2811,13 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, o
 		s.tctx.L().Warn("found error when get sql_mode from binlog status_vars", zap.Error(err))
 	}
 
+	qec.timezone, err = event.GetTimezoneByStatusVars(ev.StatusVars, s.upstreamTZ.String())
```

Comment:
I'm a bit afriad that the string converted from s.upstreamTZ is not the same as the original string which builds s.upstreamTZ. So maybe we need to add s.upstreamTZStr to store the original string.

Can you find some material to check it?

Reply:
Added s.upstreamTZStr.